src/createServerClient.ts

```diff
@@ -28,115 +116,94 @@ export function createServerClient<
+        headers: {
+          ...options?.global?.headers,
+          "X-Client-Info": `supabase-ssr/${VERSION}`,
+        },
+      },
+      auth: {
+        ...(options?.cookieOptions?.name
+          ? { storageKey: options.cookieOptions.name }
+          : null),
+        ...options?.auth,
+        flowType: "pkce",
+        autoRefreshToken: false,
+        detectSessionInUrl: false,
+        persistSession: true,
+        storage,
       },
     },
-    auth: {
-      flowType: "pkce",
-      autoRefreshToken: isBrowser(),
-      detectSessionInUrl: isBrowser(),
-      persistSession: true,
-      storage: {
-        // to signal to the libraries that these cookies are coming from a server environment and their value should not be trusted
-        isServer: true,
-        getItem: async (key: string) => {
-          const chunkedCookie = await combineChunks(
-            key,
-            async (chunkName: string) => {
-              if (typeof cookies.get === "function") {
-                return await cookies.get(chunkName);
-              }
-            },
-          );
-          return chunkedCookie;
-        },
-        setItem: async (key: string, value: string) => {
-          if (typeof cookies.set === "function") {
-            // first delete all chunks so that there would be no overlap
-            await deleteAllChunks(key);
-
-            const chunks = createChunks(key, value);
-
-            for (let i = 0; i < chunks.length; i += 1) {
-              const chunk = chunks[i];
+  );
 
-              await cookies.set(chunk.name, chunk.value, {
-                ...DEFAULT_COOKIE_OPTIONS,
-                ...cookieOptions,
-                maxAge: DEFAULT_COOKIE_OPTIONS.maxAge,
-              });
-            }
-          }
-        },
-        removeItem: async (key: string) => {
-          if (
-            typeof cookies.remove === "function" &&
-            typeof cookies.get !== "function"
-          ) {
-            console.log(
-              "Removing chunked cookie without a `get` method is not supported.\n\n\tWhen you call the `createServerClient` function from the `@supabase/ssr` package, make sure you declare both a `get` and `remove` method on the `cookies` object.\n\nhttps://supabase.com/docs/guides/auth/server-side/creating-a-client",
-            );
-            return;
-          }
+  client.auth.onAuthStateChange(async (event) => {
+    // The SIGNED_IN event is fired very often, but we don't need to
+    // apply the storage each time it fires, only if there are changes
+    // that need to be set -- which is if setItems / removeItems have
+    // data.
+    const hasStorageChanges =
+      Object.keys(setItems).length > 0 || Object.keys(removedItems).length > 0;
 
-          await deleteAllChunks(key);
+    if (
+      hasStorageChanges &&
+      (event === "SIGNED_IN" ||
+        event === "TOKEN_REFRESHED" ||
+        event === "USER_UPDATED" ||
+        event === "SIGNED_OUT")
+    ) {
```

Comment:
@kangmingtay This is the change so that PKCE sign in works in server side. (The code exchange for the session.)